cmd/serve/serve.go

```diff
@@ -229,14 +236,13 @@ func setupListeners(listenClientUrls []string) (listeners map[string]net.Listene
 
 // Serve starts the GRPC server on the provided socketPath. If the port is non-zero, it listens
 // to the TCP port instead of the unix domain socket.
-func Serve(log *zap.Logger, listenClientUrls []string, s server.Observer) error {
+func Serve(log *zap.Logger, listenClientUrls []string, s *server.LocalObserverServer) error {
```

Comment:
This swap seems like it's headed in the wrong direction. We used to accept an interface, and now we're taking a pointer to a concrete struct.

Reply:
it used to take an interface because LocalObserverServer and ObserverServer implemented the same interface with the duplicated code.

now there is only one implementation of local observer server, so i didn't bother to keep the interface. i can define an interface for the local observer server if you think that might be useful. let me know.